CHANGELOG.md

```diff
@@ -1,3 +1,7 @@
+# v0.5.1
+
+- [#678](https://github.com/helmholtz-analytics/heat/pull/678) Bugfix: Many implicit array transfers on DNDarray initialization
```

Comment:
Can you be more descriptive? If I get it right, we are avoiding implicit array transfers or replacing them with ...

(Apologies for this comment, having spent so much time trying to figure out what our `Pending additions` for 0.5.0 meant, I'm a bit traumatized 😅 )

Reply:
Possible change for clarity

Bugfix: Internal functions now use explicit device parametes for ``DNDarray`` and ``torch.Tensor`` initializations.